middleware/ordered_group.go

```diff
@@ -76,18 +117,24 @@ func (g *orderedIDs) Swap(id string, m ider) (ider, error) {
 	if len(id) == 0 {
 		return nil, fmt.Errorf("swap from ID must not be empty")
 	}
-	if len(m.ID()) == 0 {
+
+	iderID := m.ID()
+	if len(iderID) == 0 {
 		return nil, fmt.Errorf("swap to ID must not be empty")
 	}
 
-	if err := g.order.Swap(id, m.ID()); err != nil {
+	if g.IsSlot(id) && id != iderID {
+		return nil, fmt.Errorf("swap to ID must match slot being swapped")
+	}
```

Comment:
Would be good to update the doc for this function calling out the new error case.